distributed/cli/dask_worker.py

```diff
@@ -107,6 +122,32 @@ def main(scheduler, host, worker_port, http_port, nanny_port, nthreads, nprocs,
         logger.error("Failed to launch worker.  You cannot use the --name argument when nprocs > 1.")
         exit(1)
 
+    if contact_addr and not listen_addr:
+        logger.error("Failed to launch worker. "
+                     "Must specify --listen-addr when --contact-addr is given")
+        exit(1)
```

Comment:
This check should maybe be `(not contact_attr) != (not listen_addr)` for symmetry